examples/image_processing/image_view.py

```diff
@@ -1,49 +1,48 @@
-from pydm.PyQt.QtCore import QObject, pyqtSlot, pyqtSignal
-from pydm.widgets.channel import PyDMChannel
 from pydm import Display
-import numpy as np
+import threading
 from os import path
 from skimage.feature import blob_doh
 from marker import ImageMarker
 from pyqtgraph import mkPen
 
+
 class ImageViewer(Display):
+
     def __init__(self, parent=None, args=None):
         super(ImageViewer, self).__init__(parent=parent, args=args)
-        self.image_channel = "ca://MTEST:TwoSpotImage"
-        self.image_width_channel = "ca://MTEST:ImageWidth"
-        self.ui.imageView.widthChannel = self.image_width_channel
-        self.ui.imageView.image_channel = ""
-        self.markers = []
-        
+        self.markers_lock = threading.Lock()
+        self.ui.imageView.process_image = self.process_image
+        self.ui.imageView.newImageSignal.connect(self.draw_markers)
+        self.markers = list()
+        self.blobs = list()
+
     def ui_filename(self):
         return 'image_view.ui'
-        
+
     def ui_filepath(self):
         return path.join(path.dirname(path.realpath(__file__)), self.ui_filename())
-    
-    @pyqtSlot(np.ndarray)
-    def new_image_received(self, new_waveform):
-        #Reshape the 1D waveform into 2D
-        img = new_waveform.reshape((int(512),-1), order='C')
-        #Find blobs in the image with scikit-image
-        blobs= blob_doh(img, max_sigma=512, min_sigma=64, threshold=.02)
-        #Remove any existing blob markers        
+
+    def draw_markers(self, *args, **kwargs):
+        self.markers_lock.acquire()
```

Comment:
Use the `lock` as `contextmanager here. 

https://docs.python.org/3/library/threading.html#using-locks-conditions-and-semaphores-in-the-with-statement